scene/resources/visual_shader.cpp

```diff
@@ -898,6 +899,31 @@ VisualShader::VaryingType VisualShader::get_varying_type(const String &p_name) {
 	return varyings[p_name].type;
 }
 
+void VisualShader::set_preview_shader_parameter(const String &p_name, const Variant &p_value) {
+	if (p_value.get_type() == Variant::NIL) {
+		if (preview_params.has(p_name)) {
+			preview_params.erase(p_name);
+		} else {
+			return;
+		}
+	} else {
+		if (preview_params.has(p_name) && preview_params[p_name] == p_value) {
```

Comment:
Here you could use `getptr()` or `find()` to avoid double lookup, but I guess it doesn't matter that much.